商品情報編集機能

miracletea opened the pull request:
# What
商品情報編集機能の作成

#Why
商品情報編集機能を実装のため

【Gyazo】
・ログイン状態の出品者は、商品情報編集ページに遷移できる動画
https://gyazo.com/adedb693fee8b3742b1de85298f37e32

 ・必要な情報を適切に入力して「更新する」ボタンを押すと、商品の情報を編集できる動画
https://gyazo.com/44a1a07b3fdebe00d9739f306d6efadd

 ・入力に問題がある状態で「変更する」ボタンが押された場合、情報は保存されず、編集ページに戻りエラーメッセージが表示される動画
https://gyazo.com/096388e143fad247b29f7b708d00f033

 ・何も編集せずに「更新する」ボタンを押しても、画像無しの商品にならない動画
https://gyazo.com/581f27ab5c8bc142556760c78ec03c23

 ・ログイン状態の場合でも、URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
https://gyazo.com/3a80dfe4d946764210539e46d8f63ac4

 ・ログアウト状態の場合は、URLを直接入力して商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
https://gyazo.com/cf4b1b043b66f8ba51b4078bbe4f0265

 ・商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示される動画（商品画像・販売手数料・販売利益に関しては、表示されない状態で良い）
https://gyazo.com/29ab90b0d249a003fde086d5ee3ff35d

miracletea committed:
editアクションの実装

miracletea committed:
updateアクションの実装

TakashiKozuma commented on app/controllers/items_controller.rb:
```diff
@@ -23,12 +23,26 @@ def show
+  end
+
+  def update
+    @item = Item.find(params[:id])
```
```Rb
@item = Item.find(params[:id])
```
23,27,31,41行目で定義している@itemの記述が重複しているようです。
こちらはprivate以下にメソッド化して、beforeアクションで呼び出す形にすることでリファクタリングしましょう。
そうすることで複数回使用する変数をメソッド化することで可読性が上がり、改修作業が発生した際も手を加えやすくなります。

TakashiKozuma reviewed (commented):
レビュー依頼ありがとうございます！
コメントを致しましたので、ご確認お願いいたします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

miracletea committed:
beforeによるコードのまとめ

app/controllers/items_controller.rb
  end

  def update
    @item = Item.find(params[:id])
@TakashiKozuma TakashiKozuma 19 minutes ago
@item = Item.find(params[:id])
23,27,31,41行目で定義している@itemの記述が重複しているようです。
こちらはprivate以下にメソッド化して、beforeアクションで呼び出す形にすることでリファクタリングしましょう。
そうすることで複数回使用する変数をメソッド化することで可読性が上がり、改修作業が発生した際も手を加えやすくなります。

miracletea commented:
ご確認ありがとうございます。
コメント内容を反映しました、宜しくお願い致します。

maimai0401 commented on app/controllers/items_controller.rb:
```diff
@@ -20,15 +22,29 @@ def create
-      # redirect_to action: :index
-    # end
-  # end
+  def move_to_new_user_session
```
「ログイン状態のユーザーであっても、URLを直接入力して出品していない商品の商品情報編集ページへ遷移しようとすると、トップページに遷移すること」という実装条件を満たす為の記述に修正しましょう。

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください。
どうぞよろしくおねがいします。

miracletea committed:
ログインユーザー以外の商品詳細遷移の制限を追加

app/controllers/items_controller.rb
      # redirect_to action: :index
    # end
  # end
  def move_to_new_user_session
@maimai0401 maimai0401 1 hour ago
「ログイン状態のユーザーであっても、URLを直接入力して出品していない商品の商品情報編集ページへ遷移しようとすると、トップページに遷移すること」という実装条件を満たす為の記述に修正しましょう。

miracletea commented:
ご確認ありがとうございます。
ページ遷移の制限を修正しました、宜しくお願い致します。

TakashiKozuma commented:
追加のレビュー依頼ありがとうございます！
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
